Sources/XcodeProj/Project/XcodeProj.swift

```diff
@@ -133,74 +148,122 @@ extension XcodeProj: Writable {
+        XcodeProj.schemesPath(path) + "xcschememanagement.plist"
     }
 
     /// Returns debugger folder path relative to the given path.
     ///
-    /// - Parameter path: `.xcodeproj` file path
-    /// - Parameter schemeName: scheme name
+    /// - Parameter path: parent folder of debugger folder (xcshareddata or xcuserdata)
     /// - Returns: debugger folder path relative to the given path.
     public static func debuggerPath(_ path: Path) -> Path {
-        XcodeProj.sharedDataPath(path) + "xcdebugger"
+        path + "xcdebugger"
     }
 
     /// Returns breakpoints plist path relative to the given path.
     ///
-    /// - Parameter path: `.xcodeproj` file path
-    /// - Parameter schemeName: scheme name
+    /// - Parameter path: parent folder of debugger folder (xcshareddata or xcuserdata)
     /// - Returns: breakpoints plist path relative to the given path.
     public static func breakPointsPath(_ path: Path) -> Path {
         XcodeProj.debuggerPath(path) + "Breakpoints_v2.xcbkptlist"
     }
 
-    /// Writes all project breakpoints to the given path.
+    /// Writes all schemes to the given path.
     ///
-    /// - Parameter path: path to `.xcodeproj` file.
+    /// - Parameter path: parent folder of schemes folder (xcshareddata or xcuserdata)
+    /// - Parameter override: if schemes should be overridden. Default is true.
+    ///   If true will remove all existing schemes before writing.
+    ///   If false will throw error if scheme already exists at the given path.
+    public static func writeSchemes(schemes: [XCScheme], path: Path, override: Bool = true) throws {
+        let schemesPath = XcodeProj.schemesPath(path)
+        if override, schemesPath.exists {
+            try schemesPath.delete()
+        }
+
+        try schemesPath.mkpath()
+        for scheme in schemes {
+            try scheme.write(path: XcodeProj.schemePath(path, schemeName: scheme.name), override: override)
+        }
+    }
+
+    /// Writes breakpoints to the given path.
+    ///
+    /// - Parameter path: parent folder of debugger folder (xcshareddata or xcuserdata)
     /// - Parameter override: if project should be overridden. Default is true.
     ///   If true will remove all existing debugger data before writing.
     ///   If false will throw error if breakpoints file exists at the given path.
-    public func writeBreakPoints(path: Path, override: Bool = true) throws {
-        guard let sharedData = sharedData else { return }
-
+    public static func writeBreakPoints(breakpoints: XCBreakpointList?, path: Path, override: Bool = true) throws {
         let debuggerPath = XcodeProj.debuggerPath(path)
         if override, debuggerPath.exists {
             try debuggerPath.delete()
         }
+
         try debuggerPath.mkpath()
-        try sharedData.breakpoints?.write(path: XcodeProj.breakPointsPath(path), override: override)
+        try breakpoints?.write(path: XcodeProj.breakPointsPath(path), override: override)
     }
```

Comment:
What is the reason this logic is not part of `XCBreakpointList.write`? Is it because it's dealing with the parent (`xcdebugger`) folder and not the responsibility of `XCBreakpointList`? 

We could wrap `XCBreakpointList` in a new `XCDebugger` type, something like : 

```
public final class XCDebugger: Equatable, Writable {
    public let breakpointList: XCBreakpointList

    public init(path: Path) throws {
        breakpointList = try XCBreakpointList(path: XCDebugger.breakPointsPath(path))
    }

    public init(breakpointList: XCBreakpointList) {
        self.breakpointList = breakpointList
    }

    // MARK: - Writable

    public func write(path: Path, override: Bool) throws {
        if override, path.exists {
            try path.delete()
        }
        try path.mkpath()
        try breakpointList.write(path: XCDebugger.breakPointsPath(path), override: override)
    }

    /// Returns breakpoints plist path relative to the given path.
    ///
    /// - Parameter path: path to debugger folder
    /// - Parameter schemeName: scheme name
    /// - Returns: breakpoints plist path relative to the given path.
    public static func breakPointsPath(_ path: Path) -> Path {
        path + "Breakpoints_v2.xcbkptlist"
    }

    // MARK: - Equatable

    public static func == (lhs: XCDebugger, rhs: XCDebugger) -> Bool {
        lhs.breakpointList == rhs.breakpointList
    }
}
```
To move the writing logic out of XcodeProj.swift. But that would introduce a breaking change..

Reply:
Nice find! Doesn’t that mainly describe why the breakpoints are written independently of the other shared data, but not why those breakpoints would not be wrapped inside of a XCDebugger object? That would still allow the debugger files to be written independently (and in practice they’ll contain only the breakpoints). Anyway, it would be a breaking change..